src/libraries/System.Private.CoreLib/src/System/DateOnly.cs

```diff
@@ -0,0 +1,904 @@
+            }
+
+            return new DateOnly(DayNumberFromDateTime(result.parsedDate));
+        }
+
+        /// <summary>
+        /// Converts the specified span representation of a date to its DateOnly equivalent using the specified format.
+        /// The format of the string representation must match the specified format exactly or an exception is thrown.
+        /// </summary>
+        /// <param name="s">A span containing the characters that represent a date to convert.</param>
+        /// <param name="format">A span containing the characters that represent a format specifier that defines the required format of s.</param>
+        /// <returns>An object that is equivalent to the date contained in s, as specified by format.</returns>
+        public static DateOnly ParseExact(ReadOnlySpan<char> s, ReadOnlySpan<char> format) => ParseExact(s, format, DateTimeFormatInfo.CurrentInfo, DateTimeStyles.None);
+
+        private const string OFormat = "yyyy'-'MM'-'dd";
+        private const string RFormat = "ddd, dd MMM yyyy";
+
+        /// <summary>
+        /// Converts the specified span representation of a date to its DateOnly equivalent using the specified format, culture-specific format information, and style.
+        /// The format of the string representation must match the specified format exactly or an exception is thrown.
+        /// </summary>
+        /// <param name="s">A span containing the characters that represent a date to convert.</param>
+        /// <param name="format">A span containing the characters that represent a format specifier that defines the required format of s.</param>
+        /// <param name="provider">An object that supplies culture-specific formatting information about s.</param>
+        /// <param name="style">A bitwise combination of enumeration values that indicates the permitted format of s. A typical value to specify is None.</param>
+        /// <returns>An object that is equivalent to the date contained in s, as specified by format, provider, and style.</returns>
+        public static DateOnly ParseExact(ReadOnlySpan<char> s, ReadOnlySpan<char> format, IFormatProvider? provider, DateTimeStyles style = DateTimeStyles.None)
+        {
+            if ((style & ~DateTimeStyles.AllowWhiteSpaces) != 0)
+            {
+                throw new ArgumentException(SR.Argument_InvalidDateStyles, nameof(style));
+            }
+
+            if (format.Length == 1)
+            {
+                if (format[0] == 'o' || format[0] == 'O')
+                {
+                    format = OFormat;
+                    provider = CultureInfo.InvariantCulture.DateTimeFormat;
+                }
+                else if (format[0] == 'r' || format[0] == 'R')
+                {
+                    format = RFormat;
+                    provider = CultureInfo.InvariantCulture.DateTimeFormat;
+                }
```

Comment:
```suggestion
                switch (format[0])
                {
                    case 'o':
                    case 'O':
                        format = OFormat;
                        provider = CultureInfo.InvariantCulture.DateTimeFormat;
                        break;
                        
                    case 'r':
                    case 'R':
                        format = RFormat;
                        provider = CultureInfo.InvariantCulture.DateTimeFormat;
                        break;
                }
```

Reply:
I'll change that but I am wondering is that for perf reason. or just coding style?

Reply:
In this case, it's more around style and being consistent with the similar checks elsewhere in the DateTime code, which are done with a switch.

(From a perf perspective, it does give the C# compiler flexibility that it lacks if you use if/then.  With if/then, the compiler must do the checks exactly in the order you specified: that could be good if you know that o/O are _the_ most common and should be checked first, but it makes r/R more expensive.  The compiler has the flexibility to implement the switch in a variety of ways.  This is more relevant when there are more cases, though.)